### ISSUE TITLE
git master HEAD: compile error with gcc 7.1.0

### ISSUE THREAD

**Issue by tonycurtis** [openucx/ucx#1638]
```
make[2]: Entering directory '/home/tc/src/ucx/master/build/src/uct'
  CC       ib/base/libuct_la-ib_md.lo
../../../source/src/uct/ib/base/ib_md.c: In function ‘uct_ib_query_md_resources’:
../../../source/src/uct/ib/base/ib_md.c:960:44: error: ‘%s’ directive output may be truncated writing up to 63 bytes into a region of size 13 [-Werror=format-truncation=]
     snprintf(md_name, UCT_MD_NAME_MAX, "%s/%s", UCT_IB_MD_PREFIX, device->name);
                                            ^~
../../../source/src/uct/ib/base/ib_md.c:960:5: note: ‘snprintf’ output between 4 and 67 bytes into a destination of size 16
     snprintf(md_name, UCT_MD_NAME_MAX, "%s/%s", UCT_IB_MD_PREFIX, device->name);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../source/src/uct/ib/base/ib_md.c: In function ‘uct_ib_md_open’:
../../../source/src/uct/ib/base/ib_md.c:960:44: error: ‘%s’ directive output may be truncated writing up to 63 bytes into a region of size 13 [-Werror=format-truncation=]
     snprintf(md_name, UCT_MD_NAME_MAX, "%s/%s", UCT_IB_MD_PREFIX, device->name);
                                            ^~
../../../source/src/uct/ib/base/ib_md.c:960:5: note: ‘snprintf’ output between 4 and 67 bytes into a destination of size 16
     snprintf(md_name, UCT_MD_NAME_MAX, "%s/%s", UCT_IB_MD_PREFIX, device->name);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by jeffhammond** [comment]
It would be ideal if you would backport this fix to 1.2.x

**Comment by alinask** [comment]
PR for v1.2.x - https://github.com/openucx/ucx/pull/1776
